lib/file_database.rb

```diff
@@ -16,6 +16,17 @@ def database_empty?
     all.empty?
   end
 
+  def create(contact)
+    file_array = all.push(contact)
+
+    file.open << JSON.generate(file_array)
+    file.flush
```

Comment:
you don't need to reopen the file here, as the file will be given to you already open.

In the example from the previous pull request, you would do this in your CLI app:

```ruby
file = File.open('path/to/file', 'r')
database = FileDatabase.new(file)
```

Think of the user interface.
The `$stdin` and `$stdout` streams are already open, you don't have to open them again.
In the same way, the file stream remains open for the whole time the application is running.


One of the points from this ticket:
> We always wipe out the file and write the full contents to it.

and [in the ticket to familiarize yourself with File](https://github.com/Matt-Warnock/contact-book/issues/44), it says:
> We can also `rewind` files, or we can `truncate` them to wipe their contents. 

so if we go to [the documentation for the `truncate` method](https://ruby-doc.org/core-3.0.0/File.html#method-c-truncate), it says:
> Truncates the file file_name to be at most integer bytes long.

So we can do:

```ruby
file.truncate(0)
file.write(all.push(contact))
file.flush
```

Reply:
if I do this does it not wipe the file before I can read the existing contacts on it?